lib/agent.js

```diff
@@ -205,6 +205,10 @@ Agent.prototype.addSpanFilter = function (fn) {
   this._spanFilters.add(fn)
 }
 
+function isInternalFrame (frame) {
+  return /^node_modules\/elastic-apm/.test(frame.filename)
```

Comment:
While this works, shouldn't it technically be checking for the complete module name: `elastic-apm-node`?

To support Windows we should also use `path.join()`.

And I'm not sure if this is still the case or just an old habit of mine, but I usually define RegExes outside functions so they can be compiled once and reused many times. It used to be that V8 couldn't optimize this part. But maybe that's no longer an issue

Reply:
By using only part of the name, it can catch `elastic-apm-http-client` too. I can move the regex out too. 👍

Reply:
Good point. Didn't think of `elastic-apm-http-client`

Reply:
I've externalized the regex. Do you think for Windows paths it's sufficient to just check for _either_ `/` or `\`?

Reply:
You could also just change the regex to:

```js
const internalFrameRegex = new RegExp(path.join('^node_modules', ' elastic-apm'))
```

Reply:
Hmm just ran a test and it seems that the `\` in the windows path is being interpreted as an escape char in the regex

Reply:
... so I guess we'll either need to double escape them or do the thing you suggested - which might be easier at this point

Reply:
Oh, sorry if it was unclear: I wasn't implying it _had_ support for Windows paths, I was asking if changing it to just understand both was what it needed. 😅

Reply:
Yeah, got that. I just then tried to investigate if just using `path.join` would be a simpler way to do it, but turns out it wasn't. So allowing it to accept both is probably easiest.